views/analysis.py

```diff
@@ -544,6 +547,23 @@ def validate_analysis_extent(request):
         return HttpResponseServerError()
 
 
+def calculate_area(request):
+    if request.method != 'POST':
+        return HttpResponseBadRequest()
+
+    if 'area' in request.session:
+        area = request.session['area']
```

Comment:
> when investigating this, i guess it's because the run_analysis on main.html redirect the page to a new page. 
> so, all the values are gone.

Yes it is a new page, but you will have hazard_id, exposure_id and analysis_extent from Jinja2 context, because you will have `analysis` object. You can use this information to request `validate_analysis_extent` to get the new area.